In Termux login_tty() is implemented in the libutil package to support android-21.

diff -u -r /home/fornwall/lib/android-ndk/sysroot/usr/include/utmp.h ./usr/include/utmp.h
--- /home/fornwall/lib/android-ndk/sysroot/usr/include/utmp.h	2017-06-20 17:41:56.000000000 +0200
+++ ./usr/include/utmp.h	2017-07-02 11:42:29.724986159 +0200
@@ -102,11 +102,6 @@
 void endutent(void);
 
 
-#if __ANDROID_API__ >= 23
-int login_tty(int) __INTRODUCED_IN(23);
-#endif /* __ANDROID_API__ >= 23 */
-
-
-__END_DECLS
+int login_tty(int);
 
 #endif /* _UTMP_H_ */
